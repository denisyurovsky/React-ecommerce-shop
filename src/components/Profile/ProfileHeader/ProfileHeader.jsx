import PropTypes from 'prop-types';
import React, { useState, useEffect } from 'react';

import Breadcrumbs from '../../../components/Breadcrumbs/Breadcrumbs';
import { Title } from '../../../components/Title/Title';
import { LINKS } from '../../../constants/linkConstants';

import styles from './ProfileHeader.module.scss';

const { HOME, USERS, PROFILE } = LINKS;
const links = [HOME, USERS];

function ProfileHeader({ profile, title }) {
  const [header, setHeader] = useState('');

  useEffect(() => {
    if (profile) {
      const { id, firstName, lastName } = profile;

      links[2] = {
        url: `${USERS.url}/${id}`,
        text: `${firstName} ${lastName}`,
      };

      return setHeader(`Profile of ${firstName} ${lastName}`);
    }

    links[2] = PROFILE;
    setHeader(title);
  }, [profile, title]);

  return (
    <>
      <Breadcrumbs links={links} />

      <div className={styles.header}>
        <Title>{header}</Title>
      </div>
    </>
  );
}

ProfileHeader.propTypes = {
  title: PropTypes.string,
  profile: PropTypes.shape({
    id: PropTypes.number.isRequired,
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }),
};

export default ProfileHeader;
