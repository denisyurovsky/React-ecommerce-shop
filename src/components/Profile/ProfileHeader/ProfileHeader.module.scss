.header {
  align-self: flex-start;
}
