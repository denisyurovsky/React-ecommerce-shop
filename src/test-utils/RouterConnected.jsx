import { createMemoryHistory } from 'history';
import PropTypes from 'prop-types';
import React from 'react';
import { Router, Route, Routes } from 'react-router-dom';

function RouterConnected({ component: Component, path, initialPaths }) {
  const history = createMemoryHistory({ initialEntries: initialPaths });

  return (
    <Router location={history.location} navigator={history}>
      <Routes>
        <Route path={path} element={Component} />
      </Routes>
    </Router>
  );
}

RouterConnected.propTypes = {
  component: PropTypes.object.isRequired,
  path: PropTypes.string,
  initialPaths: PropTypes.arrayOf(PropTypes.string),
};

RouterConnected.defaultProps = {
  path: '/',
  initialPaths: ['/'],
};

export default RouterConnected;
