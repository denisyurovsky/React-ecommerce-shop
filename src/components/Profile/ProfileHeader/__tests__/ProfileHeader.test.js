import { render, screen } from '@testing-library/react';
import React from 'react';

import RouterConnected from '../../../../test-utils/RouterConnected';
import ProfileHeader from '../ProfileHeader';

const profile = {
  id: 1,
  firstName: 'John',
  lastName: 'Smith',
};

describe('ProfileHeader component', () => {
  describe('snapshots', () => {
    it('renders a valid snapshot', () => {
      const { asFragment } = render(
        <RouterConnected component={<ProfileHeader profile={profile} />} />
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('should show header text', () => {
      render(
        <RouterConnected component={<ProfileHeader profile={profile} />} />
      );
      expect(screen.getByText(/Profile of John Smith/i)).toBeInTheDocument();
    });
  });
});
