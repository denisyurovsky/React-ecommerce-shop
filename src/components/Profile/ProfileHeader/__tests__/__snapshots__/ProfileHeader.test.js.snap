// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileHeader component snapshots renders a valid snapshot 1`] = `
<DocumentFragment>
  <div
    class="breadcrumbs"
    role="presentation"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="breadcrumb"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="breadcrumb"
            href="/users"
          >
            Users
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="breadcrumb"
            href="/users/1"
          >
            John Smith
          </a>
        </li>
      </ol>
    </nav>
  </div>
  <div
    class="header"
  >
    <h2
      class="header"
      data-testid="title"
    >
      Profile of John Smith
    </h2>
  </div>
</DocumentFragment>
`;
